mcs/restapi/sessions/sessions.go

```diff
@@ -0,0 +1,93 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package sessions
+
+import (
+	"crypto/rand"
+	"io"
+	"strings"
+	"sync"
+
+	mcCmd "github.com/minio/mc/cmd"
+)
+
+type Singleton struct {
+	sessions map[string]*mcCmd.Config
+}
+
+var instance *Singleton
+var once sync.Once
+
+// Returns a Singleton instance that keeps the sessions
+func GetInstance() *Singleton {
+	once.Do(func() {
+		//build sessions hash
+		sessions := make(map[string]*mcCmd.Config)
+
+		instance = &Singleton{
+			sessions: sessions,
+		}
+	})
+	return instance
+}
+
+func (s *Singleton) NewSession(cfg *mcCmd.Config) (string, error) {
+	sessionId := RandomCharString(64)
+	// check if this sessions doesn't have a collision
+	if _, ok := s.sessions[sessionId]; ok == false {
+		s.sessions[sessionId] = cfg
+	} else {
+		// if there's a collision, try again
+		return s.NewSession(cfg)
+	}
+	return sessionId, nil
+}
+
+func (s *Singleton) ValidSession(sessionId string) bool {
+	if _, ok := s.sessions[sessionId]; ok == false {
+		return true
+	}
+	return false
+}
+
```

Comment:
can you please add tests for all this functions?

Reply:
done